src/lib/navigation/base-navigation-mode-factory.provider.ts

```diff
@@ -0,0 +1,72 @@
+import {FreeNavigationMode} from './free-navigation-mode';
+import {NavigationMode} from './navigation-mode.interface';
+import {SemiStrictNavigationMode} from './semi-strict-navigation-mode';
+import {StrictNavigationMode} from './strict-navigation-mode';
+import {WizardComponent} from '../components/wizard.component';
+import {NavigationModeInput} from './navigation-mode-input.interface';
+import {NavigationModeFactory} from './navigation-mode-factory.interface';
+
+/**
+ * A factory used to create [[NavigationMode]] instances
+ */
+export class BaseNavigationModeFactory implements NavigationModeFactory {
```

Comment:
The new classes should be added as exports to the [index.ts](https://github.com/madoar/angular-archwizard/blob/develop/src/index.ts) file.

Reply:
I have added both `NavigationModeFactory` and `BaseNavigationModeFactory` to `index.ts`.

Reply:
Oh yes, sorry I've missed that!